lib/modules/onc_program/bulk_data_group_validation_sequence.rb

```diff
@@ -344,63 +343,68 @@ def log_and_reraise_if_error(request, response, truncated)
       end
 
       def streamed_ndjson_get(url, headers)
-        ctx = OpenSSL::SSL::SSLContext.new
-        ctx.verify_mode = OpenSSL::SSL::VERIFY_PEER # set globally to VERIFY_NONE if disable_verify_peer set
-        ctx.set_params unless OpenSSL::SSL::VERIFY_PEER == OpenSSL::SSL::VERIFY_NONE
-        response = HTTP.headers(headers).get(url, ssl_context: ctx)
+        # We don't want to keep a huge log of everything that came through,
+        # but we also want to show up to a reasonable number.
+        recent_lines = []
+        line_count = 0
 
-        # We need to log the request, but don't know what will be in the body
-        # until later.  These serve as simple summaries to get turned into
-        # logged requests.
+        next_block = String.new
 
         request_for_log = {
           method: 'GET',
           url: url,
           headers: headers
         }
 
+        response_block = proc { |response|
+          response.read_body do |chunk|
+            next_block << chunk
+            resource_list = next_block.lines
+
+            response_for_log = {
+              code: response.code,
+              headers: response.header,
+              body: String.new,
+              truncated: false
+            }
+
+            # Skip processing the last line since the it may not be complete (still appending from stream)
+            next_block = resource_list.pop
+            # Skip if the last_line is empty
+            # Cannot use .blank? since docker-compose complains "invalid byte sequence in US-ASCII" during unit test
+            next_block = String.new next_block unless next_block.nil? || next_block.strip.empty?
```

Comment:
Does this line actually do anything? To me it looks like it sometimes assigns a copy of `next_block` to itself, and I don't see how that would be useful.

Reply:
Is it because Ruby complains now if you mutate strings by default (unless you provide some kind of directive on the top of the file to allow that)?  So we create a new copy of the string?  I recall having to add that kind of stuff here, though I haven't looked too closely to see if this is really related to that.

Reply:
I removed it, as discussed in standup last week.